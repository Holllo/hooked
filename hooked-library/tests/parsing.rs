use std::path::PathBuf;

use {color_eyre::Result, hooked_library::Config};

use {insta::assert_debug_snapshot, test_case::test_case};

#[test_case("defaults" ; "defaults")]
#[test_case("with-hooks" ; "with-hooks")]
fn test_config_parsing(file_name: &str) -> Result<()> {
  let file = PathBuf::from(format!("tests/parsing/{file_name}.toml"));
  let config = Config::from_toml_file(file)?;
  assert_debug_snapshot!(format!("{file_name}"), config);
  Ok(())
}
