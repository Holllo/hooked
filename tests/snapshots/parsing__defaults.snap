---
source: tests/parsing.rs
expression: config
---
Config {
    general: General {
        directory: "hooks",
    },
}
